src/ts/components/forms/button.tsx

```diff
@@ -16,37 +16,39 @@ export interface ButtonProps extends ComponentProps, HTMLProps<HTMLElement> {
    * Make the button small
    */
   small?: boolean;
-}
+} & React.ButtonHTMLAttributes<HTMLElement> &
+  OptionalComponentPropAndHTMLAttributes;
 
 /**
  * Used in place of a standard `button` tag, this component adds additional styles and effects.
  */
-export class Button extends PureComponent<ButtonProps, {}> {
-  public render() {
-    const {
-      children,
-      className,
-      block,
-      large,
-      small,
-      component: Component = 'button',
-      ...remainingProps
-    } = this.props;
+const Button = (props: ButtonProps) => {
+  const {
+    children,
+    className,
+    block,
+    large,
+    small,
+    component: Component = 'button',
+    ...remainingProps
+  } = props;
+
+  const myClassNames = [
+    'button',
+    block ? 'block' : null,
+    small ? 'small' : null,
+    large ? 'large' : null,
+    className,
+  ];
 
-    const myClassNames = [
-      'button',
-      block ? 'block' : null,
-      small ? 'small' : null,
-      large ? 'large' : null,
-      className,
-    ];
+  // Cast necessary otherwise types are too complex
+  const CastComponent = Component as 'button';
```

Comment:
I guess this does open us up to some potential type issues here but it usually still complains if they are incompatible right? :)

Reply:
It'll complain if the props are incompatible with button (which is the default types), but not for other props. So if someone's set the `component="div"` and passes in a `type="submit"` neither the consumer, nor Roe will error. Unfortunately this is a side-effect of using an automatic doc generator - it doesn't understand anything except the most basic of types, so we can't really do it in a more sensible way (yet).